src/main/kotlin/app/batch/MessageProducer.kt

```diff
@@ -0,0 +1,49 @@
+package app.batch
+
+import app.domain.DataKeyResult
+import app.domain.EncryptionResult
+import com.beust.klaxon.JsonObject
+import org.apache.commons.lang3.StringUtils
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class MessageProducer {
+    fun produceMessage(jsonObject: JsonObject,
+                       encryptionResult: EncryptionResult,
+                       dataKeyResult: DataKeyResult,
+                       database: String,
+                       collection: String): String {
+        val id = jsonObject.obj("_id")?.toJsonString()!!
+        val modified = jsonObject.obj("_lastModifiedDateTime")
+        val date = modified?.get("\$date")
+        val dateStr = if (date != null) date as String else ""
+        val type = jsonObject.get("@type") ?: "MONGO_UPDATE"
+        return if (StringUtils.isNotBlank(dateStr)) {
+                """{
+                |   "message": {
+                |       "@type": "$type",
+                |       "_id": $id,
+                |       "_lastModifiedDateTime": "$date",
+                |       "collection" : "$collection",
+                |       "db": "$database",
+                |       "dbObject": "${encryptionResult.encrypted}",
+                |       "encryption": {
+                |           "keyEncryptionKeyId": "${dataKeyResult.dataKeyEncryptionKeyId}",
+                |           "initialisationVector": "${encryptionResult.initialisationVector}",
+                |           "encryptedEncryptionKey": "${dataKeyResult.ciphertextDataKey}"
+                |       }
+                |   }
+                |}""".trimMargin()
+        }
+        else {
+            logger.error("No '_lastModifiedDateTime' in record '$id' from '$database/$collection'.")
+            ""
```

Comment:
should these characters be here?

Reply:
Yeah it returns an empty string if it doesn't have the required inputs - that's the contract agreed with the Uday's work in progress.

Reply:
Oh it's that damn Kotlin return thing. really not obvious what that is at all!